### PR TITLE
gh actions: add workflow to build tool

### PR DESCRIPTION
**Pull request by shajmakh**
Add new workflow to build and release binary for tools/hugepages-machineconfig-generator to enable using it externally.

### CODE REVIEW

**Comment by shajmakh** [comment]
need to add a new tag from master to enable this workflow to run correctly and provide releases with the tool binary.

File: .github/workflows/release-hp-mc-generator.yml [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+        with:
+          fetch-depth: 0 
+
+      - name: build machineconfig generator tool for hugepages
+        run: make dist-hugepages-mc-genarator
+
+      - name: release the tool binary 
+        uses: ncipollo/release-action@v1
+        with:
+          name: hugepagesmcgenartor
```
[the remark this answers is not in the document]
Comment (shajmakh) [reply .github/workflows/release-hp-mc-generator.yml:40]: correct! will modify it, thanks!

**Review by fromanirh — COMMENTED** [review]
/lgtm
/hold
adding gh actions is a good thing in general, and so is having the PAO tools easily available.
We expect to make few runs in general and in this repo in particular - being on maintenance mode
Low risk, high gain.

**Comment by fromanirh** [comment]
> /retest-required

it is good to have a full green run, but for this specific PR we can safely override failing lines. Adding github actions config can't possibly break any production flow (or test code)

**Comment by fromanirh** [comment]
> /override ci/prow/e2e-gcp-operator-upgrade /override ci/prow/e2e-gcp

don't worry, I can override once we reach consensus :) we only need more LGTMs

**Pull request merged**